torch/csrc/tensor/python_tensor.cpp

```diff
@@ -256,15 +280,27 @@ static bool PyTensorType_Check(PyObject* obj) {
   return it != tensor_types.end();
 }
 
+static PyTensorType *get_tensor_type(THPDtype *obj) {
+  auto it = std::find_if(tensor_types.begin(), tensor_types.end(),
+    [obj](const PyTensorType& x) {
+      return x.dtype == obj;
+    });
+  return it == tensor_types.end() ? nullptr : &(*it);
+}
+
 void set_default_tensor_type(const at::Type& type) {
   set_type(*default_tensor_type, type.backend(), type.scalarType());
 }
 
 void py_set_default_tensor_type(PyObject* obj) {
-  if (!PyTensorType_Check(obj)) {
+  PyTensorType *type;
+  if (PyTensorType_Check(obj)) {
+    type = (PyTensorType*)obj;
+  } else if (THPDtype_Check(obj)) {
+    type = get_tensor_type((THPDtype*)obj);
```

Comment:
This can end up setting `type` to nullptr which is dereferenced below.

Reply:
Good point; I'll change get_tensor_type to throw an exception because that should never happen.